packages/web-app-files/tests/unit/views/SharedViaLink.spec.js

```diff
@@ -1,29 +1,194 @@
+import { shallowMount, mount } from '@vue/test-utils'
+import flushPromises from 'flush-promises'
+import { getStore, localVue, createFile } from './views.setup.js'
+import MixinFilesListPagination from '@files/src/mixins/filesListPagination'
+import FileActions from '@files/src/mixins/fileActions'
+import MixinFilesListPositioning from '@files/src/mixins/filesListPositioning'
+import SharedViaLink from '@files/src/views/SharedViaLink.vue'
+
+const $route = {
+  params: {
+    fileId: '2147491323'
+  },
+  meta: {
+    title: 'Resolving private link'
+  }
+}
+
+const activeFiles = [createFile({ id: 1234, type: 'file' })]
+
+localVue.prototype.$client.requests = {
+  ocs: jest.fn(() =>
+    Promise.resolve({
+      status: 200,
+      json: jest.fn(() =>
+        Promise.resolve({
+          ocs: {
+            data: activeFiles
+          }
+        })
+      )
+    })
+  )
+}
+
+const stubs = {
+  'router-link': true,
+  'no-content-message': true,
+  pagination: true,
+  'oc-table-files': true,
+  'context-actions': true,
+  'list-info': true
+}
+
+const selectors = {
+  listLoader: 'list-loader-stub',
+  noContentMessage: '#files-shared-via-link-empty',
+  ocTableFiles: '#files-shared-via-link-table',
+  contextActions: 'context-actions-stub',
+  listInfo: 'list-info-stub'
+}
+
 describe('SharedViaLink view', () => {
+  jest
+    .spyOn(MixinFilesListPagination.methods, '$_filesListPagination_updateCurrentPage')
+    .mockImplementation()
+  const spyAdjustTableHeaderPosition = jest.spyOn(
+    MixinFilesListPositioning.methods,
+    'adjustTableHeaderPosition'
+  )
+  const spyTriggerDefaultAction = jest
+    .spyOn(FileActions.methods, '$_fileActions_triggerDefaultAction')
+    .mockImplementation()
+  const spyLoadResources = jest.spyOn(SharedViaLink.methods, 'loadResources')
+  const spyRowMounted = jest.spyOn(SharedViaLink.methods, 'rowMounted')
+
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
   describe('when the page has loaded successfully', () => {
-    it.todo('should load the resources')
-    it.todo('should adjust the table header position')
+    beforeEach(async () => {
+      getShallowWrapper()
+      await flushPromises()
+    })
+
+    it('should load the resources', () => {
+      expect(spyLoadResources).toHaveBeenCalledTimes(1)
+    })
+    it('should adjust the table header position', () => {
+      expect(spyAdjustTableHeaderPosition).toHaveBeenCalledTimes(1)
+    })
   })
 
   describe('when the view is still loading', () => {
-    it.todo('should show list-loader component')
+    it('should show list-loader component', () => {
+      const wrapper = getShallowWrapper()
+      const listLoader = wrapper.find(selectors.listLoader)
+
+      expect(listLoader.exists()).toBeTruthy()
+      expect(listLoader).toMatchSnapshot()
```

Comment:
It feels a bit odd testing snapshot of a child component in the parent instead of tests of that component...